### PR TITLE
Support `includeSortVector` option for `find` and `findOne` command

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Support `includeSortVector` option to return the vector used for ANN search for `find` and `findOne` command.

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java [hunk 1]
```diff
@@ -129,7 +138,8 @@ public static FindOperation unsorted(
    * @param logicalExpression expression contains filters and their logical relation
    * @param projection projections, see FindOperation#projection
    * @param readType type of the read
-   * @param objectMapper object mapper to use
+   * @param objectMapper object mapper to use * @param vector vector to search * @param
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java:141]: Looks like multiple `@param` docs ended up on the same line here?

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java [hunk 2]
```diff
@@ -584,7 +584,44 @@ public void happyPath() {
...
+          .statusCode(200)
+          .body("data.documents[0]._id", is("3"))
+          .body("data.documents[0].$vector", is(notNullValue()))
+          .body("data.documents[1]._id", is("2"))
+          .body("data.documents[1].$vector", is(notNullValue()))
+          .body("data.documents[2]._id", is("1"))
+          .body("data.documents[2].$vector", is(notNullValue()))
+          .body("errors", is(nullValue()))
+          .body("status", is(notNullValue()))
+          .body("status.sortVectors", nullValue());
```
Comment (vkarpov15) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:624]: Why is this testing `sortVectors` not `sortVector`?

**Review by vkarpov15 — COMMENTED** [review]
A couple of small comments but LGTM overall

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java (hunk 2 above)
Comment (maheshrajamani) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:624]: Typo. Fixed it.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java [hunk 3]
```diff
@@ -59,14 +62,16 @@ public record FindOperation(
...
    * @return FindOperation for a single document unsorted find
    */
   public static FindOperation unsortedSingle(
       CommandContext commandContext,
       LogicalExpression logicalExpression,
       DocumentProjector projection,
       ReadType readType,
-      ObjectMapper objectMapper) {
+      ObjectMapper objectMapper,
+      boolean includeSortVector) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java:74]: Do we still need this passed even though this is unsorted? Or in general, not just for "vsearch" case? I guess spec allows specifying `includeSortVector` for all find()/findOne() cases so perhaps that makes sense.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadOperationPage.java [hunk 4]
```diff
@@ -12,23 +15,37 @@
...
+    boolean includeSortVector,
+    float[] vector)
     implements Supplier<CommandResult> {
   @Override
   public CommandResult get() {
+    Map<CommandStatus, Object> status = null;
+    if (includeSortVector) {
+      // add sort vector to the response
+      status = new HashMap<>();
+      status.put(CommandStatus.SORT_VECTOR, vector);
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadOperationPage.java:34]: Could also use `Collections.singletonMap(CommandStatus.SORT_VECTOR, vector)`?

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndDeleteCommandResolver.java [hunk 5]
```diff
@@ -69,7 +69,8 @@ private FindOperation getFindOperation(
           DocumentProjector.includeAllProjector(),
           ReadType.DOCUMENT,
           objectMapper,
-          vector);
+          vector,
+          false);
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndDeleteCommandResolver.java:73]: So `findOneAndXxx()` can never return/include sort vector, even if one was used?

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindCommand.java [hunk 6]
```diff
@@ -61,5 +61,12 @@ public record Options(
...
               type = SchemaType.BOOLEAN,
               implementation = Boolean.class)
-          boolean includeSimilarity) {}
+          boolean includeSimilarity,
+
+      // return vector embedding used for ANN sorting
+      @Schema(
+              description = "Return vector embedding used for ANN sorting.",
+              type = SchemaType.BOOLEAN,
+              implementation = Boolean.class)
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindCommand.java:70]: Use of `Boolean.class` is probably fine for "implementation" (and used already above for "includeSimilarity") but technically type for primitive `boolean` is `Boolean.TYPE`.

**Review by tatu-at-datastax — APPROVED** [review]
LGTM

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndDeleteCommandResolver.java (hunk 5 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndDeleteCommandResolver.java:73]: Checked the spec, only `find()` and `findOne()` support it.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java (hunk 3 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java:74]: Looks like spec explicitly allows use without actual ANN so this makes sense.

**Pull request merged**